src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -59,8 +59,10 @@ int LFUDAPolicy::init(CephContext *cct, const DoutPrefixProvider* dpp, asio::io_
   };
 
   static auto block_callback = [this](
-          const DoutPrefixProvider* dpp, std::string& key, uint64_t offset, uint64_t len, std::string version, bool dirty, optional_yield y, std::string& restore_val) {
-    update(dpp, key, offset, len, version, dirty, y, restore_val);
+          const DoutPrefixProvider* dpp, std::string& key, uint64_t offset, uint64_t len, std::string version, bool dirty, optional_yield y, std::vector<std::string>& restore_vals) {
+    for (auto& val : restore_vals) {
```

Comment:
you are calling `update` method once with the localWeight value and then with the invalid flag - that will not work - in both the cases it will update the localWeight xattr value. Moreover invalid xattr belongs to objects and not blocks so you need to modify update_dirty_object() method and not `update` method.